generator/src/test/java/com/scottlogic/deg/generator/inputs/profileviolation/ProfileViolationIntegrationTests.java

```diff
@@ -0,0 +1,152 @@
+package com.scottlogic.deg.generator.inputs.profileviolation;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.ProfileFields;
+import com.scottlogic.deg.generator.Rule;
+import com.scottlogic.deg.generator.builders.AndBuilder;
+import com.scottlogic.deg.generator.builders.OrBuilder;
+import com.scottlogic.deg.generator.builders.RuleBuilder;
+import com.scottlogic.deg.generator.outputs.manifest.ManifestWriter;
+import com.scottlogic.deg.generator.violations.ViolatedProfile;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static com.scottlogic.deg.generator.inputs.profileviolation.TypeEqualityHelper.assertListProfileTypeEquality;
+import static com.shazam.shazamcrest.MatcherAssert.assertThat;
+import static com.shazam.shazamcrest.matcher.Matchers.sameBeanAs;
+
+/**
+ * Defines integration tests for all classes involved in Profile Violation.
+ */
+public class ProfileViolationIntegrationTests {
+    private IndividualRuleProfileViolator target;
+    private Path tempOutputPath;
+
+    @BeforeEach
+    public void setUp() {
+        ManifestWriter manifestWriter = new ManifestWriter();
```

Comment:
There should be 2 fixtures:
1. Integration tests, proving that all elements can be created (by IoC) and work together
1. Violation logic/framework tests that do not side-effect to disk (e.g. use a mock ManifestWriter) and simply verify that the violated profiles match the expected result (of which there will be many tests)